src/components/product.js

```diff
@@ -111,73 +114,62 @@ export default class Product extends Component {
    * @return {Object} image object.
    */
   get image() {
-    if (!this.model.selectedVariant || !this.model.selectedVariant.image || !this.model.selectedVariant.image.variants.length) {
+    if (!this.selectedVariant || (!this.selectedVariant.image && !this.options.contents.imgWithCarousel)) {
       return null;
     }
-    const availableSizes = this.model.selectedVariant.image.variants;
-
-    let imageSize = 'grande';
-
-    if (this.options.width && this.options.layout === 'vertical') {
-      imageSize = availableSizes.filter((image) => {
-        const containerWidth = parseInt(this.options.width, 10);
-        return parseInt(image.dimension, 10) >= containerWidth * 1.5;
-      })[0].name;
-    }
+    const imageSize = parseInt(this.options.width, 10) || 480;
+    let image;
+    let id;
 
-    if (this.options.imageSize) {
-      imageSize = this.options.imageSize;
-    }
+    const imageOptions = {
+      maxWidth: imageSize,
+      maxHeight: imageSize,
+    };
 
-    let sourceImage = this.model.selectedVariant.image;
     if (this.selectedImage) {
-      sourceImage = this.model.images.filter((image) => {
-        return image.id === this.selectedImage.id;
-      })[0];
+      id = this.selectedImage.id;
+      image = ShopifyBuy.Image.Helpers.imageForSize(this.selectedImage, imageOptions);
+    } else {
+      id = this.selectedVariant.image.id;
+      image = ShopifyBuy.Image.Helpers.imageForSize(this.selectedVariant.image, imageOptions);
     }
-
-    return Object.assign({}, sourceImage.variants.filter((image) => {
-      return image.name === imageSize;
-    })[0], {
-      id: sourceImage.id,
-    });
+    return {id, src: image};
   }
 
   /**
    * get formatted cart subtotal based on moneyFormat
    * @return {String}
    */
   get formattedPrice() {
-    if (!this.model.selectedVariant) {
+    if (!this.selectedVariant) {
       return '';
     }
-    return formatMoney(this.model.selectedVariant.price, this.globalConfig.moneyFormat);
+    return formatMoney(this.selectedVariant.price, this.globalConfig.moneyFormat);
   }
 
   /**
    * get formatted cart subtotal based on moneyFormat
    * @return {String}
    */
   get formattedCompareAtPrice() {
-    if (!this.model.selectedVariant) {
+    if (!this.selectedVariant) {
       return '';
     }
-    return formatMoney(this.model.selectedVariant.compareAtPrice, this.globalConfig.moneyFormat);
+    return formatMoney(this.selectedVariant.compareAtPrice, this.globalConfig.moneyFormat);
   }
 
   /**
    * get data to be passed to view.
    * @return {Object} viewData object.
    */
   get viewData() {
-    const image = Object.assign({}, this.currentImage);
     return merge(this.model, this.options.viewData, {
       classes: this.classes,
       contents: this.options.contents,
       text: this.options.text,
       optionsHtml: this.optionsHtml,
       decoratedOptions: this.decoratedOptions,
-      currentImage: image,
+      currentImage: this.currentImage,
```

Comment:
Would this be better stated as `return merge(this.model, this.options.viewData, this)`? Or do all the fields need to be explicitly enumerated?